custom-vu/src/main/kotlin/jces1209/vu/page/DcIssuePage.kt

```diff
@@ -57,4 +63,78 @@ class DcIssuePage(
     override fun linkIssue(): DcIssueLinking {
         return DcIssueLinking(driver)
     }
+
+    override fun isTimeSpentFormAppeared(): Boolean {
+        try {
+            driver
+                .wait(
+                    ExpectedConditions.presenceOfElementLocated(By.id("log-work-time-logged"))
+                )
+        } catch (e: Exception) {
+            return false
+        }
+        return true
+    }
+
+    override fun cancelTimeSpentForm(): AbstractIssuePage {
+        driver
+            .wait(
+                timeout = Duration.ofSeconds(5),
+                condition = ExpectedConditions.presenceOfElementLocated(By.id("issue-workflow-transition-cancel"))
+            )
+            .click()
+        driver
+            .wait(
+                timeout = Duration.ofSeconds(7),
+                condition = ExpectedConditions.invisibilityOfElementLocated(By.id("issue-workflow-transition-cancel"))
+            )
+        return this
+    }
+
+    override fun fillInTimeSpentForm(): AbstractIssuePage {
+        driver
+            .wait(
+                ExpectedConditions.presenceOfElementLocated(By.id("log-work-time-logged"))
+            )
+            .click()
+        Actions(driver)
+            .sendKeys("1h")
+            .perform()
+        driver
+            .wait(
+                ExpectedConditions.presenceOfElementLocated(By.id("issue-workflow-transition-submit"))
+            )
+            .click()
+        driver
+            .wait(
+                timeout = Duration.ofSeconds(7),
+                condition = ExpectedConditions.invisibilityOfElementLocated(By.id("issue-workflow-transition-cancel"))
+            )
+        return this
+    }
+
+    override fun transition(): DcIssuePage {
+        waitForPage()
+        WebDriverWait(driver, 7)
+            .until {
+                try {
+                    driver
```

Comment:
Why is it necessary to use such workaround?
Is element not clickable after `ExpectedConditions.elementToBeClickable`?

Reply:
Marked as TODO